Initial take at reworking "Queries" documentation (#1551)
edgedb/edgedb: docs/edgeql/overview.rst

   .. code-block:: edgeql

      SELECT (
        User.firstname,
        User.friends.firstname,
        User.friends.lastname,
        Issue.priority.name,
        Issue.number,
        Status.name
      );

   In the above query, the longest common prefixes are: ``User``,
   ``User.friends``, ``Issue``, and ``Status.name``.

2. Make a *query input list* of all unique set references which appear
   directly in the query (including the common path prefixes identified above).
   The set references and path prefixes in this list are called *input
   set references*,  and the sets they represent are called *input
   sets*. Order this list such that an input references come before
   any other input set reference for which it is a prefix (sorting
   lexicographically works).

3. Compute a set of *input tuples*.

   - Begin with a set containing a single empty tuple.

>    - For each input set reference, we compute a *dependent* cartesian

Reviewer: s/cartesian/Cartesian/g

     product of the input tuple set (``X``) so far and the input set
     ``Y`` being considered. In this dependent product, we pair each
     tuple ``x`` in the input tuple set ``X`` with each element of the
     subset of the input set ``Y`` corresponding to the tuple ``x``. (For
     example, in the above example, computing the dependent product
     of User and User.friends would pair each user with all of their
     friends.)

>      (Mathematically, ``X' = {(x, y) | x \in X, y \in f(x)}``, if ``f(x)``

Reviewer: Feel free to use the Unicode symbol ∈ directly

     selects the appropriate subset.)

     The set produced becomes the new input tuple set and we continue
     down the list.
   - As a caveat to the above, if an input set appears exclusively as

>      part of an :ref:`ref_eql_fundamentals_optional` argument, produce

Reviewer: ```suggestion
     an :ref:`ref_eql_fundamentals_optional` argument, produce
```

     pairs with a placeholder value ``Missing`` instead of an empty
     cartesian product in the above
     set. (Mathematically, I think the corresponds to having ``f(x) =
     {Missing}`` whenever it would otherwise produce an empty set.)

4. Iterate over the set of input tuples, and on every iteration:

   - in the query and its subqueries, replace each input set reference with the

>      corresponding value from the input tuple or an empty set if the value

Reviewer: if the value is ``Missing``

   .. code-block:: edgeql

      SELECT (
        User.firstname,
        User.friends.firstname,
        User.friends.lastname,
        Issue.priority.name,
        Issue.number,
        Status.name
      );

   In the above query, the longest common prefixes are: ``User``,
   ``User.friends``, ``Issue``, and ``Status.name``.

2. Make a *query input list* of all unique set references which appear
   directly in the query (including the common path prefixes identified above).
   The set references and path prefixes in this list are called *input
   set references*,  and the sets they represent are called *input

>    sets*. Order this list such that an input references come before

Reviewer 2: "an" seems to be not needed here, no?

```suggestion
   sets*. Order this list such that input references come before
```

   any other input set reference for which it is a prefix (sorting
   lexicographically works).

3. Compute a set of *input tuples*.

   - Begin with a set containing a single empty tuple.
   - For each input set reference, we compute a *dependent* Cartesian
     product of the input tuple set (``X``) so far and the input set
     ``Y`` being considered. In this dependent product, we pair each
     tuple ``x`` in the input tuple set ``X`` with each element of the

>      subset of the input set ``Y`` corresponding to the tuple ``x``. (For
>      example, in the above example, computing the dependent product

Reviewer 2: ```suggestion
     subset of the input set ``Y`` corresponding to the tuple ``x``. (E.g.
     in the above example, computing the dependent product
```
